File: 0001-two-sum/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hashmap: HashMap<i32, i32> = HashMap::with_capacity(nums.len());
        for (index, &value) in nums.iter().enumerate() {
            if let Some(&other_index) = hashmap.get(&(target - value)) {
                return vec![index as i32, other_index];
            }
            hashmap.insert(value, index as i32);
        }
        vec![]
    }
}

File: 0001-two-sum/solution.java
import java.util.Set;
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            int value = target - nums[i];
            if (seen.containsKey(value)) {
                return new int[]{seen.get(value), i};
            }
            seen.put(nums[i], i);
        }
        return new int[2];
    }
}
